### PR TITLE
ISPN-13067 DataSource exception sorter

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13067

### CODE REVIEW

**Comment by pruivo** [comment]
@tristantarrant checkstyle error in CI

**Comment by tristantarrant** [comment]
Fixed

File: server/runtime/src/main/java/org/infinispan/server/datasource/DataSourceFactory.java [hunk 1]
```diff
@@ -46,6 +52,19 @@ public static DataSource create(DataSourceConfiguration configuration) throws SQ
             .leakTimeout(Duration.ofMillis(configuration.leakDetection()))
             .reapTimeout(Duration.ofMinutes(configuration.idleRemoval()));
 
+      if (configuration.driver().contains("postgresql")) {
+         pool.exceptionSorter(new PostgreSQLExceptionSorter());
+      } else if (configuration.driver().contains("mysql")) {
+         pool.exceptionSorter(new MySQLExceptionSorter());
```
Comment (ryanemerson) [inline server/runtime/src/main/java/org/infinispan/server/datasource/DataSourceFactory.java:58]: We can also use `MySQLExceptionSorter` for `mariadb` as they have the same error codes.

Comment (tristantarrant) [reply server/runtime/src/main/java/org/infinispan/server/datasource/DataSourceFactory.java:59]: Done

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request merged**